addon/components/ember-wrapper.js

```diff
@@ -0,0 +1,31 @@
+import React from 'npm:react';
+
+/**
+ * A React component that is used to render HTML Nodes.
+ *
+ * The primary usage of this component is to support `props.children` in `react-component`.
+ * The catch is that an additional `<span>` is introduced as a wrapper.
+ *
+ * Note that although this is possible, it should be used as a tool to migrate Ember to React
+ * without the hard requirement to start with leaf components. It is highly recommended to
+ * have clean React component tree whenever possible.
+ *
+ * Integration guide: https://reactjs.org/docs/integrating-with-other-libraries.html
+ */
+class EmberWrapper extends React.Component {
+  componentDidMount() {
+    this.$el = window.$(this.el);
+    this.$el.append(...this.props.nodes);
+  }
+
+  componentWillUnmount() {
+  }
+
+  render() {
+    return React.createElement('span', {
```

Comment:
What if you don't want a span wrapper and want a block element wrapper instead, i.e. `div`?

This could be solved with an API like this:

```handlebars
{{#react-component "the-wrapper" yieldWrapperTagName="span"}}
  <button>Test</button>
  {{value}}
{{/react-component}}
```

Or possibly we require that when react-component yields that all of its children are explicitly wrapped in a wrapper component.

```handlebars
{{#react-component "the-wrapper"}}
  {{#react-component/yield-wrapper tagName="span"}}
    <button>Test</button>
    {{value}}
  {{//react-component/yield-wrapper}}
{{/react-component}}
```

Reply:
I have thought of upgrading to React 16 so that we [can return multiple nodes](https://reactjs.org/docs/react-component.html#fragments) then we don't need the base element at all. I still need to experiment it.